src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -757,16 +754,13 @@ internal static void UnregisterMemoryLoadChangeNotification(Action notification)
 
 #endif
             }
-            else if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
-            {
-                ThrowHelper.ThrowInvalidTypeWithPointersNotSupported(typeof(T));
-            }
 
             GC_ALLOC_FLAGS flags = GC_ALLOC_FLAGS.GC_ALLOC_ZEROING_OPTIONAL;
```

Comment:
```suggestion
            // Runtime overrides GC_ALLOC_ZEROING_OPTIONAL if the type contains references, so we don't need to worry about that.
            GC_ALLOC_FLAGS flags = GC_ALLOC_FLAGS.GC_ALLOC_ZEROING_OPTIONAL;
```
I think this is more appropriate place for this comment.